modules/table/src/test/java/org/apache/ignite/internal/table/criteria/SqlSerializerTest.java

```diff
@@ -0,0 +1,110 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.criteria;
+
+import static org.apache.ignite.table.criteria.Criteria.and;
+import static org.apache.ignite.table.criteria.Criteria.columnValue;
+import static org.apache.ignite.table.criteria.Criteria.equalTo;
+import static org.apache.ignite.table.criteria.Criteria.in;
+import static org.apache.ignite.table.criteria.Criteria.not;
+import static org.apache.ignite.table.criteria.Criteria.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+/**
+ * SQL generation test.
+ */
+class SqlSerializerTest {
+    @Test
+    void testEquals() {
+        var ser = new SqlSerializer.Builder()
+                .tableName("test")
+                .columns(Set.of("A"))
+                .where(columnValue("a", equalTo("a")))
+                .build();
+
+        assertEquals("SELECT * FROM test WHERE A = ?", ser.toString());
+        assertArrayEquals(new Object[] {"a"}, ser.getArguments());
+    }
+
+    @Test
+    void testNonEquals() {
+        var ser = new SqlSerializer.Builder()
+                .tableName("test")
+                .columns(Set.of("A"))
+                .where(not(columnValue("a", equalTo("a"))))
+                .build();
+
+        assertEquals("SELECT * FROM test WHERE NOT A = ?", ser.toString());
+        assertArrayEquals(new Object[] {"a"}, ser.getArguments());
+    }
+
+    @Test
+    void testIn() {
+        var ser = new SqlSerializer.Builder()
+                .tableName("test")
+                .columns(Set.of("A"))
+                .where(not(columnValue("a", in("a", "b", "c"))))
+                .build();
+
+        assertEquals("SELECT * FROM test WHERE NOT A IN (?, ?, ?)", ser.toString());
+        assertArrayEquals(new Object[] {"a", "b", "c"}, ser.getArguments());
+    }
+
+    @Test
+    void testAnd() {
+        var ser = new SqlSerializer.Builder()
+                .tableName("test")
+                .columns(Set.of("A", "B"))
+                .where(and(columnValue("a", nullValue()), not(columnValue("b", in("a", "b", "c")))))
```

Comment:
I've tried the following condition:
`.where(columnValue("a", equalTo(in(columnValue("a", nullValue()), columnValue("b", nullValue())))))`
Unfortunately no any error and I have result as `a = ?`
It look as incorrect

Reply:
This problem occurred due to https://github.com/apache/ignite-3/pull/2810#discussion_r1386303866.
Replaced `Object` with `Comparable` in argument and append another `equalsTo` method with byte[] as argument.

Reply:
also need to have for all supported types